clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1484,6 +1495,16 @@ void KernelObjVisitor::visitArray(const CXXRecordDecl *Owner, FieldDecl *Field,
 
   if (Field->hasAttr<SYCLRequiresDecompositionAttr>()) {
     visitComplexArray(Owner, Field, ArrayTy, Handlers...);
+  } else if (AnyTrue<HandlerTys::VisitInsideSimpleContainersWithPointer...>::
+                 Value) {
+    assert(!Field->hasAttr<SYCLGenerateNewTypeAttr>() &&
+           "Arrays should trigger decomposition");
+    // We are currently in PointerHandler visitor, which implies this is a
+    // 'simple' array i.e. one that does not include special types or pointers.
+    // Array of pointers/ array of type containing pointers will be handled in
+    // a follow-up PR. Currently, they continue to trigger decomposition, and
+    // will be handled in 'if' statment above.
```

Comment:
```suggestion
    // FIXME: Array of pointers/ array of type containing pointers
  // will be handled in a follow up PR. Currently, they continue
  // to trigger decomposition.
```

Reply:
Thanks for catching the typo here! I fixed the typo but did not add the "FIXME" so that it reads better in the paragraph.